src/rest.cpp

```diff
@@ -590,6 +590,57 @@ static bool rest_chaininfo(const std::any& context, HTTPRequest* req, const std:
     }
 }
 
+RPCHelpMan getdeploymentinfo();
+
+static bool rest_deploymentinfo(const std::any& context, HTTPRequest* req, const std::string& strURIPart)
+{
+    if (!CheckWarmup(req))
+        return false;
+
+    std::string hashStr;
+    const RESTResponseFormat rf = ParseDataFormat(hashStr, strURIPart);
+
+    switch (rf) {
+    case RESTResponseFormat::JSON: {
+        JSONRPCRequest jsonRequest;
+        jsonRequest.context = context;
+
+        if (!hashStr.empty()) {
+            uint256 hash;
+            if (!ParseHashStr(hashStr, hash))
+                return RESTERR(req, HTTP_BAD_REQUEST, "Invalid hash: " + hashStr);
+
+            LOCK(cs_main);
+            const uint256 hash_bi(ParseHashV(hashStr, "blockhash"));
+            ChainstateManager* maybe_chainman = GetChainman(context, req);
+            if (!maybe_chainman) return false;
+            ChainstateManager& chainman = *maybe_chainman;
+            const CBlockIndex* blockindex = chainman.m_blockman.LookupBlockIndex(hash_bi);
+            if (!blockindex) {
+                return RESTERR(req, HTTP_BAD_REQUEST, "Block not found");
+            }
+            
+            UniValue block_hash(UniValue::VOBJ);
+            block_hash.pushKV("blockhash", hashStr);
+        
+            jsonRequest.params = block_hash;
+        } else {
+            jsonRequest.params = UniValue(UniValue::VARR);
```

Comment:
Could move this up after `jsonRequest.context = context;`, and just pushKV directly

Reply:
Nice, thanks!